src/jdk.crypto.cryptoki/share/classes/sun/security/pkcs11/P11SecretKeyFactory.java

```diff
@@ -99,6 +99,9 @@ static long getPKCS11KeyType(String algorithm) {
 
     // returns direct lookup result of keyTypes using algorithm
     static long getKeyType(String algorithm) {
+        if ("ChaCha20-Poly1305".equals(algorithm)) {
+            algorithm = "ChaCha20";
+        }
```

Comment:
Have you considered adding "ChaCha20-Poly1305" into the "keyTypes" instead? Then you don't need to add the if-block here.

Reply:
That would indeed be cleaner solution. I'll make this change. Thank you.

Reply:
Please update the 2nd copyright year to 2022 also.